provider/kubernetes/client.go

```diff
@@ -264,3 +252,15 @@ func eventHandlerFunc(events chan<- interface{}, obj interface{}) {
 	default:
 	}
 }
+
+// translateNotFoundError will translate a "not found" error to a boolean return
+// value which indicates if the resource exists and a nil error.
+func translateNotFoundError(err error) (bool, error) {
+	if err != nil {
+		if kubeerror.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
```

Comment:
I think we should add a quick unit test for this function... it's easy to get some boolean wrong while the test should be easy to write.